sakura/sakura/sakura.py

```diff
@@ -21,41 +22,51 @@ class Sakura:
     def __init__(
         self,
         providers: Optional[dict[str, Provider]],
+        pubsub_clients: Optional[dict[str, PubSubClient]],
         loggers: Optional[list[Logger]] = None,
     ):
         self._once_functions = []
         self.__providers = providers
+        self.__pubsub = pubsub_clients
         self.__loggers = loggers
         self.__tasks: list[typing.Coroutine] = []
         self.__should_exit = False
         self.__force_exit = False
-        self.init_logging()
 
-    def init_logging(self):
+    def setup_loggers(self):
         for logger in self.loggers:
             logger.setup()
 
-        config = functools.reduce(merge_dicts, (a.get_basic_config() for a in self.loggers))
+        config = functools.reduce(
+            merge_dicts,
+            (a.get_basic_config() for a in self.loggers),
+        )
```

Comment:
Use dynaconf merge?

Reply:
this is not related to dynaconf, every logger according to the current implementation returns a dict that represents python's default logger syntax config,
this is actually a code from the original sakura code base, and not related to the current branch, and the loggers mechanism will be refactored as soon as I finish with this issue.